2014-10-14  Paul Pluzhnikov  <ppluzhnikov@google.com>

	[BZ #12926]
	* sysdeps/unix/sysv/linux/check_pf.c (make_request): Avoid
	infinite loop when __recvmsg returns 0.

diff --git a/sysdeps/unix/sysv/linux/check_pf.c b/sysdeps/unix/sysv/linux/check_pf.c
index c7fd9b0..976f249 100644
--- a/sysdeps/unix/sysv/linux/check_pf.c
+++ b/sysdeps/unix/sysv/linux/check_pf.c
@@ -180,7 +180,7 @@ make_request (int fd, pid_t pid)
 	};
 
       ssize_t read_len = TEMP_FAILURE_RETRY (__recvmsg (fd, &msg, 0));
-      if (read_len < 0)
+      if (read_len <= 0)
 	goto out_fail;
 
       if (msg.msg_flags & MSG_TRUNC)
